src/olympia/yara/tasks.py

```diff
@@ -0,0 +1,52 @@
+import yara
+
+from django.conf import settings
+
+import olympia.core.logger
+
+from olympia.amo.celery import task
+from olympia.files.models import FileUpload
+from olympia.files.utils import SafeZip
+
+from .models import YaraResult
+
+log = olympia.core.logger.getLogger('z.yara.task')
+
+
+@task
+def run_yara(upload_pk):
```

Comment:
I know the whole repo is lacking in that department, but a docstring would be nice.

And in fact it may be a good place to document that this task can *only* work as part of the submission process - because otherwise it's too late, we keep old `FileUpload`s for a little while but we move the xpi as soon as we create the version. It tripped me up for a second when I tried to re-run the task on an add-on I had created earlier.